src/AppInstallerRepositoryCore/Microsoft/ARPHelper.h

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "Microsoft/SQLiteIndex.h"
+#include <AppInstallerArchitecture.h>
+#include <winget/Registry.h>
+#include <winget/ManifestInstaller.h>
+#include <wil/resource.h>
+
+#include <string>
+
+namespace AppInstaller::Repository::Microsoft
+{
+    // A helper to find the various locations that contain ARP (Add/Remove Programs) entries.
+    struct ARPHelper
+    {
+        // See https://docs.microsoft.com/en-us/windows/win32/msi/uninstall-registry-key for details.
+        std::wstring SubKeyPath{ L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall" };

```

Comment:
>std::wstring [](start = 8, length = 12)

Make these wstrings const?
Or to be consistent across the project, use static const wchar_t* ?

Reply:
The registry APIs require null termination, and string_view does not guarantee that (even though one constructed from a constant would probably (maybe?) have that).  We could create our own zstring_view, which is not uncommon, but given the nature of these I figured I could simply sit them in the struct that exists only as long as it needs to.

---
In reply to: [518406200](https://github.com/microsoft/winget-cli/pull/633#discussion_r518406200) [](ancestors = 518406200)